graphs/growth.py

```diff
@@ -36,55 +35,82 @@ def TG_formula(TG, A):
-            'dash': 'dash',
-            'width': 3
-        },
-    ))
-
-# offline.plot(figure, filename='graph.html')
-figure.show()
+
+class CountryData:
+    def __init__(self, path, name, country_basic):
+        self.name = name
+        self.country_basic = country_basic
+        self.path = os.path.join(path, 'data-' + name + '.yaml')
+        with open(self.path, 'r') as stream:
+            try:
+                data = yaml.safe_load(stream)
+                self.active = [point['positive'] - point['recovered'] - point['dead'] for point in data]
+                self.dead = [point['dead'] for point in data]
+                self.recovered = [point['recovered'] for point in data]
+                self.date_list = [point['date'] for point in data]
+            except yaml.YAMLError as exc:
+                raise exc
+
+        self.cumulative_active = list(filter(lambda x: x >= self.country_basic.case_count, itertools.accumulate(self.active)))
+        self.date_list = self.date_list[len(self.active) - len(self.cumulative_active):]
+        self.x = np.arange(1, len(self.cumulative_active) * 2)
+        self.y = self.country_basic.formula.lambd(self.x)
+
+        self.last_date = datetime.strptime(self.date_list[-1], '%Y-%m-%d')
+        self.date_list += [(self.last_date + timedelta(days=d)).strftime('%Y-%m-%d') for d in range(1, 51)]
+
+    def create_country_figure(self):
+    
+        layout = Layout(title=f"Active cases in {self.country_basic.name}",
+                        xaxis=dict(autorange=True,
+                                title=r'$\text{Days since the 200}^\mathrm{th}\text{ case}$'),
+                        yaxis=dict(type='log', autorange=True, title='COVID-19 active cases'),
+                        hovermode='x',
+                        font={'size': 15},
+                        legend=dict(x=0.01, y=0.99, borderwidth=1))
+
+        figure = Figure(layout=layout)
+        figure.add_trace(
+            Scatter(x=self.x,
+                    y=self.cumulative_active,
+                    mode='lines+markers',
+                    name=f"Active cases",
+                    line={'width': 3},
+                    marker={'size': 8}))
+
+        figure.add_trace(
+            Scatter(
+                x=self.x,
+                y=self.y,
+                text=self.date_list,
+                mode='lines',
+                name=self.country_basic.formula.text,
+                line={
+                    'dash': 'dash',
+                    'width': 3
+                },
+            ))
+        return figure
+
+if args.country != "ALL":
+    country = next(c for c in countries if c.name == args.country)
+    country_data = CountryData(args.data, args.country, country)
+    country_data.create_country_figure().show()
+    # offline.plot(country_data.create_country_figure(), filename='graph.html')
+    exit(0)
+
+countries_data = [CountryData(args.data, country.name, country) for country in countries]
+
+import dash
```

Comment:
Importy sa dávajú vždy na začiatok súboru, nie?

Reply:
Môže byť :) Bolo to len tak logicky oddelené.